### DISCUSSION TITLE
通过反代挂载的webdav无法进行重命名/移动操作

### DISCUSSION THREAD

**Discussion by flashlab** [AlistGo/alist#3302] [Q&A]
### 问题描述
我的alist放在威联通的NAS上，局域网通过类似`http://192.168.1.2:5244/dav/public`的url可以对里面的文件正常重命名或者移动位置。
但通过自带的apache反代之后就不行了，其他操作（新建、删除、上传、下载）是OK的，此时的url类似`https://alist.DOMAIN.com:12345/dav/public`。
在win10、android上的mixplorer上测试都是类似的情况，而且威联通自带的webdav也是如此。日志里的报错信息如下：
```
ERRO[2023-02-07 22:47:39] MOVE /dav/public/initNet.sh webdav: invalid destination
[GIN] 2023/02/07 - 22:47:39 | 502 |     160.557µs | 240e:388:8c52:6e00:79ec:637c:207a:bd9f | MOVE     "/dav/public/initNet.sh"
```
apache的配置如下：
```
<VirtualHost *:12345>
    ServerName alist.DOMAIN.com
    SSLEngine on
    SSLCipherSuite EECDH+CHACHA20:EECDH+AES
    SSLProtocol All -SSLv2 -SSLv3 -TLSv1 -TLSv1.1
    SSLCertificateFile "/etc/stunnel/stunnel.pem"
    SSLCertificateChainFile "/etc/stunnel/uca.pem"
    AllowEncodedSlashes NoDecode
    ProxyPass / http://localhost:5244/ connectiontimeout=360 timeout=360
    ProxyPassReverse / http://localhost:5244/
</VirtualHost>
```
### 可能的原因
apache没有正确返回修改后的路径，导致重命名失败.

网上搜到的多是使用nginx的解决方法，apache如何配置有没有大佬指点一二呢？
会不会跟tls有关还没测试，也许可以用http协议试一下

**Comment by Wait845** [comment]
使用nginx的解决方法后还是有同样的问题，感觉应该是alist的问题

**Comment by jkoor** [comment]
alist 配置文件中域名写反代后的域名

**Reply by flashlab** [reply]
这个改过的，但问题依然存在

**Reply by Wait845** [reply]
> 这个改过的，但问题依然存在

我安装官方文档推荐的nginx配置试了一下，没问题了

**Reply by flashlab** [reply]
具体是什么字段起作用了呢？apache问题依然存在

**Reply by Wait845** [reply]
> 具体是什么字段起作用了呢？apache问题依然存在

不清楚是什么字段，忘记备份之前的配置了.猜测可能是:X-Forwarded-Proto

**Reply by bianyukun1213** [reply]
在我的案例里，是 `X-Forwarded-Host` 字段，原来值是 `$host`，文件和文件夹无法重命名，按照官方文档设置成 `$http_host` 就可以了。

**Comment by flashlab** [comment]
加上`ProxyPreserveHost On` 貌似就正常了，不知道是新版做了修正还是apache配置正确

**Reply by ZSXB2468** [reply]
实测，是加上这个才正常，注释掉就不行了
